src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -202,39 +202,50 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                     if (hasCertificateAndKey)
                     {
                         bool hasCertReference = false;
-                        try
+                        if (sslAuthenticationOptions.CertificateContext != null && sslAuthenticationOptions.CertificateContext.IntermediateCertificates != null)
```

Comment:
Why allow CertificateContext to be null here?  If one of the older methods is used you could just move the legacy code to earlier.

Reply:
For now, the CertificateContext is only used for server and this code is used for both sides. The other aspect is that I did not won't to generate chain upfront on platforms where it is not needed (like windows)  So even if CertificateContext is not null, IntermediateCertificates may unless we add more platform logic there.

Reply:
I decided to do it also for client certificates and duplicate chain logic from here.